From 6404653130f7d8e5ddea6b9d9f9f7d193e86c2c1 Mon Sep 17 00:00:00 2001
From: Juraj Fiala <jurf@riseup.net>
Date: Thu, 10 Jan 2019 10:21:32 +0100
Subject: [PATCH] Install the AppData file

Since the AppData is useful outside the AppImage context too (GNOME
Software, Flatpak, etc.), add it to classic installations as well.
---
 appimage-build.sh                                            | 5 -----
 ...ncthingtk.appdata.xml => me.kozec.syncthingtk.appdata.xml | 0
 setup.py                                                     | 1 +
 3 files changed, 1 insertion(+), 5 deletions(-)
 rename scripts/me.kozec.syncthingtk.appdata.xml => me.kozec.syncthingtk.appdata.xml (100%)

diff --git a/appimage-build.sh b/appimage-build.sh
index ae1290cc..61944b51 100755
--- a/appimage-build.sh
+++ b/appimage-build.sh
@@ -1,6 +1,5 @@
 #!/bin/bash
 APP="syncthing-gtk"
-APPID="me.kozec.syncthingtk"
 EXEC="syncthing-gtk"
 [ x"$BUILD_APPDIR" == "x" ] && BUILD_APPDIR=$(pwd)/appimage
 
@@ -133,10 +132,6 @@ sed -i "s/Icon=.*/Icon=${APP}/g" ${BUILD_APPDIR}/${APP}.desktop
 cp -H icons/${APP}.png ${BUILD_APPDIR}/${APP}.png
 [ -e "${BUILD_APPDIR}/usr/share/${APP}/icons/${APP}.png" ] || ln -s "../../../../${APP}.png" "${BUILD_APPDIR}/usr/share/${APP}/icons/${APP}.png"
 
-# Copy appdata.xml
-mkdir -p ${BUILD_APPDIR}/usr/share/metainfo/
-cp scripts/${APPID}.appdata.xml ${BUILD_APPDIR}/usr/share/metainfo/${APPID}.appdata.xml
-
 # Copy AppRun script
 cp scripts/appimage-AppRun.sh ${BUILD_APPDIR}/AppRun
 chmod +x ${BUILD_APPDIR}/AppRun
diff --git a/scripts/me.kozec.syncthingtk.appdata.xml b/me.kozec.syncthingtk.appdata.xml
similarity index 100%
rename from scripts/me.kozec.syncthingtk.appdata.xml
rename to me.kozec.syncthingtk.appdata.xml
diff --git a/setup.py b/setup.py
index d0bca472..1da36c80 100755
--- a/setup.py
+++ b/setup.py
@@ -95,6 +95,7 @@ def find_mos(parent, lst=[]):
 		('share/icons/hicolor/64x64/emblems', glob.glob("icons/emblem-*.png") ),
 		('share/pixmaps', ["icons/syncthing-gtk.png"]),
 		('share/applications', ['syncthing-gtk.desktop'] ),
+		('share/metainfo', ['me.kozec.syncthingtk.appdata.xml'] ),
 	] + [
 		(
 			'share/icons/hicolor/%sx%s/apps' % (size,size),
